ops/__init__.py

```diff
@@ -321,3 +321,46 @@
 # rather than a runtime concern.
 
 from .version import version as __version__
+
+
+class _Main:
+    def __call__(
+        self, charm_class: Type[charm.CharmBase], use_juju_for_storage: Optional[bool] = None
+    ):
+        return _main.main(charm_class=charm_class, use_juju_for_storage=use_juju_for_storage)
+
+    def main(
+        self, charm_class: Type[charm.CharmBase], use_juju_for_storage: Optional[bool] = None
+    ):
+        return _legacy_main.main(
+            charm_class=charm_class, use_juju_for_storage=use_juju_for_storage
+        )
+
+
+main = _Main()
+"""Set up the charm and dispatch the observed event.
+
+The event name is based on the way this executable was called (argv[0]).
+
+.. jujuremoved:: 4.0
+    The ``use_juju_for_storage`` argument is not available from Juju 4.0
+
+Args:
+    charm_class: the charm class to instantiate and receive the event.
+    use_juju_for_storage: whether to use controller-side storage. If not specified
+        then Kubernetes charms that haven't previously used local storage and that
+        are running on a new enough Juju default to controller-side storage,
+        otherwise local storage is used.
+
+Recommended usage:
+
+.. code-block:: python
+
+    import ops
+
+
+    class SomeCharm(ops.CharmBase):
+        ...
+
+    ops.main(SomeCharm)
```

Comment:
Probably should have the `if __name__ == '__main__'` guard here?

Reply:
done